### PR TITLE
Matts lecture

### PR DESCRIPTION
**Pull request by MA-Penner**
### What

my lecture 
### Why

so i don't fail
### Where

in student lectures 

### Additional Notes

not sure if i did this right si fingers crossed

### CODE REVIEW

File: site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+Introduction to Search Spaces
+----------------------------
+
+Search Space Terminology 
+------------------------
+* search space size #S
+* number of (optimal) solutions #S' ,solution density #S'/#S
+* search space diameter (:math:`G_{N}`) = maximal distance between any two candidate solutions
+* N(:math:`pi`)  is the  neighbourhood relation of a given candaicae slution (:math:`pi`)
```
Comment (jameshughes89) [inline site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst:13]: did you want this to be p_i? It is fine if not; I'm just asking.

File: site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+* g(:math:`pi`) is the fitress of a given candaicae slution (:math:`pi`)
```
Comment (jameshughes89) [inline site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst:14]: fitress

File: site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+![ Local vs GlobalSmooth vs Jagged Comparison](Example-of-local-and-global-solutions-in-an-optimization-problem.png)
+
+> Fig 1. from Calculation of Detector Positions for a Source Localizing Radiation Portal Monitor System Using a Modified Iterative Genetic Algorithm - Scientific Figure on ResearchGate. Available from: https://www.researchgate.net/figure/Example-of-local-and-global-solutions-in-an-optimization-problem_fig3_322270023 [accessed 8 Nov, 2023]
+
+* Local optimization focuses on finding the best solution in a small region, while global optimization seeks the best solution overall.
+* Discuss the trade-offs between exploring locally and exploiting globally.
+
+Types Search Spaces
+--------------------------------------------
```
Comment (jameshughes89) [inline site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst:26]: too long

File: site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+## Topology
+
+- **Non-degenerate (or invertible)**: A function is considered non-degenerate or invertible if assigning the same output to two different inputs is not possible. This means that each output is unique to a single input, which allows the function to be reversed.
+
+* ∀s, s₀ ∈ S : [g(s) = g(s₀) ⇒ s = s₀];
+
+- **Locally invertible**: A function is locally invertible if, within a small vicinity around any given input, it acts invertibly. That is, within this local neighborhood, each output can still be traced back to one specific input.
+
+* ∀r ∈ S : ∀s, s₀ ∈ N(r) ∪ {r} : [g(s) = g(s₀) ⇒ s = s₀];
```
Comment (jameshughes89) [inline site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst:53]: Did this render correctly?

File: site/student-lectures/Search-Space-Analysis/Search_Space_Analysis.rst [hunk 5]
```diff
@@ -0,0 +1,232 @@
+*********************
+Search Space Analysis
+*********************
+
+test
```
Comment (jameshughes89) [inline site/student-lectures/Search-Space-Analysis/Search_Space_Analysis.rst:5]: ?

**Review by jameshughes89 — CHANGES_REQUESTED** [review]
There are some comments within. 

Further, it seems that you have 2 versions of the `rst` file --- one where the name has spaces, and the other does not. Remove whichever you are not using.

File: site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst (hunk 1 above)
Comment (MA-Penner) [reply site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst]: 1.nope. the paper I saw used pi (greek letter) so I was using the same notation as them

File: site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+- **Non-neutral**: A function is non-neutral if it does not produce the same output for distinct but close inputs. In other words, nearby inputs must lead to different outputs, ensuring the function distinguishes between closely spaced inputs.
+
+
+* ∀s ∈ S : ∀s₀ ∈ N(s) : [g(s) = g(s₀) ⇒ s = s₀].
```
[the remark this answers is not in the document]
Comment (MA-Penner) [reply site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst]: some how yes. that was chat gpts doing

File: site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst (hunk 4 above)
Comment (MA-Penner) [reply site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst]: some how yes. that was chat gpts doing

File: site/student-lectures/Search-Space-Analysis/Search_Space_Analysis.rst (hunk 5 above)
Comment (MA-Penner) [reply site/student-lectures/Search-Space-Analysis/Search_Space_Analysis.rst]: my bad! forget to get rid of that

File: site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst (hunk 2 above)
Comment (MA-Penner) [reply site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst]: thanks! just installed spell check on vs code to catch everything

**Comment by jameshughes89** [comment]
There's some conflicts you need to resolve

**Pull request merged**